tests/IntegrationTest.php

```diff
@@ -73,4 +74,53 @@ public function gettingEncryptedStuffFromGoogleShouldWork()
         $this->assertTrue($connected);
         $this->assertRegExp('#^HTTP/1\.0#', $response);
     }
+
+    /** @test */
+    public function testSelfSignedRejectsIfVerificationIsEnabled()
```

Comment:
After talking to @cboden, we agree that this is probably a *bug* in certain (older) versions of PHP, which we have yet to triage further.

This test DOES work correctly on ALL PHP versions tested via Travis.

This test does NOT work on PHP 5.5.30 on Mac, but DOES work with PHP 7.? on Mac.

Note that this test failure is not related to this library at all, we're currently looking into providing a test script which highlights this issue in the underlying SSL stream wrapper.

Let's keep track of this in https://github.com/reactphp/socket-client/issues/59 instead and get this PR in :shipit: